### PR TITLE
New Scraper and NN

### PR DESCRIPTION
**Pull request by jamescowens**
This PR is the new scraper and "neural network".

### CODE REVIEW

**Comment by jamescowens** [comment]
As I figured. Travis is going to have to be troubleshot on the build. Probably missing package dependencies...

**Comment by jamescowens** [comment]
For folks looking at this PR, detailed tracking is being done on my fork's branch at 

https://github.com/jamescowens/Gridcoin-Research/projects/1

File: src/scraper_net.cpp [hunk 1]
```diff
@@ -55,7 +55,12 @@ bool CSplitBlob::RecvPart(CNode* pfrom, CDataStream& vRecv)
...
         }
     } else {
-        if(pfrom)  pfrom->Misbehaving(SCRAPER_MISBEHAVING_NODE_BANSCORE / 5);
+        if(pfrom)
+        {
+            pfrom->Misbehaving(SCRAPER_MISBEHAVING_NODE_BANSCORE / 5);
+            LogPrintf("WARNING: CSplitBlob::RecvPart: Spurious part received from %s. Adding %u banscore.",
+                     pfrom->addr.ToString(), SCRAPER_MISBEHAVING_NODE_BANSCORE / 5);
+        }
         return error("Spurious part received!");
```
Comment (tomasbrod) [inline src/scraper_net.cpp:71]: If you extended the message, please remove this error message (redundant?).

Comment (jamescowens) [reply src/scraper_net.cpp:71]: You are right.

Comment (jamescowens) [reply src/scraper_net.cpp:71]: One question on that after looking at it carefully. The error message returned is not conditioned on the current existence of the CNode pfrom, unlike the banscore part (and my new logging message). They are not exactly duplicative.

**Comment by jamescowens** [comment]
Comment to reference issues addressed by this PR...
#289, #1324 
https://github.com/gridcoin-community/Gridcoin-Tasks/issues/98

**Comment by jamescowens** [comment]
This is a comment to link issues to this PR...

#551

**Comment by jamescowens** [comment]
Ok, this is now in sync with my other tree, integrated_scraper_removalof01230124 and I am shifting back to the integrated_scraper tree, since PR #1390 was merged.

**Review by denravonska — APPROVED** [review]

**Pull request merged**